docusaurus/docs/introduction/plugin-lifecycle.md

```diff
@@ -0,0 +1,93 @@
+
+### Phase 2. Plugin discovery 
+
+Grafana starts to discover which plugins are installed by scanning directories on the file system for every `plugin.json`.
+
+### Phase 3. Plugin loading 
+
+All plugins that were discovered in the discovery phase are checked to make sure they’re valid and that they have a valid signature. Valid plugins are referred to as _verified plugins_. 
+
+### Phase 4. Backend plugin initialization
+
+For any verified plugin that has a backend, Grafana configures a backend client. Depending on the plugin's classification, one of these things will happen: 
+- **Core plugins** - the backend client is already configured and nothing more needs to be done.
+- **External plugins** - Grafana configures the backend client to use HashiCorp’s Go Plugin System over RPC.
+
+### Phase 5. Registration
+
+All verified plugins are registered in an in-memory registry. From now on, the plugin is available within Grafana and so are referred to as _registered plugins_.
+
+### Phase 6. Start backend plugin (backend plugins only)
+
+For registered external plugins that have a backend, Grafana starts to run the backend binary as a separate process using HashiCorp’s Go Plugin System over RPC. The supported plugin protocol and version is negotiated between Grafana (client) and the plugin (server) to give Grafana an understanding of the plugin's capabilities.  
+
+A Grafana backend plugin has its own separate life cycle. So long as the backend plugin is running, Grafana will make sure to restart the backend plugin in case it crashes or is killed. When Grafana is shut down, the backend processes is then terminated.  
+
+### Phase 7. Client-side loading 
+
+After Grafana has started and the HTTP API is running, Grafana users receive the server-side rendered index page containing so-called bootstrap data. This data includes the list of available plugins and a URI to a `module.js` file that Grafana uses to instantiate the plugin. 
+
+When the user interacts with a UI that requires a plugin, Grafana will _lazy load_ the plugin's `module.js` file: 
+
+- **Data-source plugins** - A data-source plugin could be loaded in more than one way. For instance, it could be loaded in the Explore page if the user selects the data source in the dropdown, or if they load a dashboard containing a plugin data source.
+
+- **App plugins** - Apps have two different loading modes: _lazy_ and _pre-load_. Lazy app plugins load only when the user accesses the App menu item directly. Pre-load app plugins load with the Grafana app and can execute code as soon as the page loads.
+
+:::note
+
+Each plugin is loaded only once but its objects are initialized multiple times. For example, a dashboard with 10 different panel plugins will load 10 plugins with an instance of each. A dashboard with 10 panels of the same plugin will load the plugin once and have 10 instances.
+
+:::
+
+## Plugin deprecation
```

Comment:
Deprecation feels separate to the rest of this as it is more about its availability within the Plugins Catalog. I would remove

Reply:
removed.